docs/notebooks/deep_gaussian_processes.pct.py

```diff
@@ -0,0 +1,354 @@
+# %% [markdown]
+# # Using deep Gaussian processes with GPflux for Bayesian optimization.
+
+# %%
+import numpy as np
+import tensorflow as tf
+
+np.random.seed(1794)
+tf.random.set_seed(1794)
+tf.keras.backend.set_floatx("float64")
+
+# %% [markdown]
+# ## Describe the problem
+# In this example, we look to find the minimum value of the two-dimensional Branin function over the hypercube $[0, 1]^2$, modified with a discontinuity. We can represent the search space using a `Box`, and plot contours of the Branin over this space.
+#
+#
+
+# %%
+import gpflow
+from trieste.objectives import (
+    michalewicz,
+    MICHALEWICZ_2_MINIMUM,
+    MICHALEWICZ_5_MINIMUM,
+)
+from trieste.objectives.utils import mk_observer
+from util.plotting_plotly import plot_function_plotly
+from trieste.space import Box
+from math import pi
+
+function = michalewicz
+F_MINIMIZER = MICHALEWICZ_2_MINIMUM
+
+search_space = Box([0, 0], [pi, pi])
+
+fig = plot_function_plotly(
+    function,
+    search_space.lower,
+    search_space.upper,
+    grid_density=100
+)
+fig.update_layout(height=400, width=400)
+fig.show()
+
+# %% [markdown]
+# ## Sample the observer over the search space
+#
+# Sometimes we don't have direct access to the objective function. We only have an observer that indirectly observes it. In _Trieste_, an observer can output a number of datasets. In our case, we only have one dataset, the objective. We can convert a function with `scaled_discts_branin`'s signature to a single-output observer using `mk_observer`.
```

Comment:
you can probably remove this, this is covered in intro notebook with EI
it also has some old refs like `scaled_discts_branin`